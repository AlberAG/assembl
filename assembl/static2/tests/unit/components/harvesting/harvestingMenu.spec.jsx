import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import HarvestingMenu from '../../../../js/app/components/harvesting/harvestingMenu';
import * as fakeData from './fakeData';

describe('harvestingMenu component', () => {
  beforeAll(() => {
    window.getSelection = () => ({
      removeAllRanges: () => {}
    });
  });
  it('should match harvestingMenu snapshot', () => {
    const { extracts } = fakeData;
    const props = {
      extracts: extracts,
      postId: '1234',
      isHarvesting: true,
      harvestingMenuPosition: 100
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<HarvestingMenu {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});