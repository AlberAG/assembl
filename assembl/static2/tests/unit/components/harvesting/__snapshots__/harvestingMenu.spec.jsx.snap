// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`harvestingMenu component should match harvestingMenu snapshot 1`] = `
<div
  className="harvesting-container"
>
  <harvestingBadge
    extractsNumber={2}
    setExtractsBoxDisplay={[Function]}
  />
</div>
`;

exports[`harvestingMenu component should match harvestingMenu without nugget action snapshot 1`] = `
<div
  className="harvesting-container"
>
  <harvestingBadge
    extractsNumber={2}
    setExtractsBoxDisplay={[Function]}
  />
</div>
`;
